### PR TITLE
Seperated variables and functions

### PR DESCRIPTION
**Pull request by Mati-AAU**

### CODE REVIEW

**Review by Casper-NS — DISMISSED** [review]
LGTM, but you need to change it for everything in the program

File: GOAT-Compiler/SymbolTableBuilder.cs [hunk 1]
```diff
@@ -52,25 +52,26 @@ private Types _processTypeOfNode(Node node)
...
         {
-            _symbolTable.AddSymbol(node.GetId().Text, _processTypeOfNode(node.GetTypes()));
+            _symbolTable.AddVariableSymbol(node.GetId().Text, _processTypeOfNode(node.GetTypes()));
         }
 
         public override void OutAParamDecl(AParamDecl node)
         {
-            _symbolTable.AddSymbol(node.GetId().Text, _processTypeOfNode(node.GetTypes()));
-            paramTypesList.Add(_symbolTable.GetSymbol(node.GetId().Text).type);
+            Types type = _processTypeOfNode(node.GetTypes());
```
Comment (S3JER) [inline GOAT-Compiler/SymbolTableBuilder.cs]: _processTypeOfNode, why _?

**Review by S3JER — APPROVED** [review]
LGTM

File: GOAT-Compiler/SymbolTableBuilder.cs (hunk 1 above)
Comment (Mati-AAU) [reply GOAT-Compiler/SymbolTableBuilder.cs]: renamed to ProcessTypeOfNode

**Pull request merged**